Assets/MixedRealityToolkit.SDK/Experimental/Dialog/Scripts/DialogShell.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Dialog
+{
+    public class DialogShell : Dialog
+    {
+        private GameObject[] twoButtonSet;
+
+        /// <summary>
+        /// This is called after the buttons are generated and
+        /// the title and message have been set.
+        /// Perform here any operations that you'd like
+        /// </summary>
+        protected override void FinalizeLayout()
+        {
+        }
+
+        protected override void GenerateButtons()
+        {
+            //Get List of ButtonTypes that should be created on Dialog
+            List<DialogButtonType> buttonTypes = new List<DialogButtonType>();
+            foreach (DialogButtonType buttonType in Enum.GetValues(typeof(DialogButtonType)))
+            {
+                if (buttonType == DialogButtonType.None)
+                {
+                    continue;
+                }
+
+                // If this button type flag is set
+                if ((buttonType & result.Buttons) == buttonType)
+                {
+                    buttonTypes.Add(buttonType);
+                }
```

Comment:
simplify to 
```c#
if (buttonType != DialogButtonType.None && result.Buttons.HasFlag(buttonType))
{
buttonTypes.add(buttonType);
}
```

https://docs.microsoft.com/en-us/dotnet/api/system.enum.hasflag?view=netframework-4.8